### ISSUE TITLE
Delete code under velox/substrait

### ISSUE THREAD

**Issue by mbasmanova** [facebookincubator/velox#8895]
### Bug description

This code is no longer used. Let's delete it to avoid confusing and cycles spent on maintanance.

CC: @kgpai 

### System information

n/a

**Comment by kgpai** [comment]
cc: @rui-mo @FelixYBW

**Comment by FelixYBW** [comment]
I'm OK to remove them since they are moved in Gluten now. See if UltraData guys still wants to work on it.

**Comment by EpsilonPrime** [comment]
I don't think anyone else can use the version that's inside Gluten.  Ideally the Substrait->Velox code would live in its own repository so it can be used by anyone needing a Substrait gateway to Velox (similar to datafusion's substrait offering).

**Comment by FelixYBW** [comment]
> I don't think anyone else can use the version that's inside Gluten. Ideally the Substrait->Velox code would live in its own repository so it can be used by anyone needing a Substrait gateway to Velox (similar to datafusion's substrait offering).

Yes, but the problem now is it's not mature yet and no one is developing or maintain the code.  I don't know if anyone still uses it. In future if someone owns the work, they can start from current code.

**Comment by drin** [comment]
Is velox open to someone else maintaining the code in a separate repository as @EpsilonPrime mentioned?

I am researching the use of substrait to span engines such as Acero, DuckDB, etc. I would be willing to be a maintainer for a separate repository for the forseeable future; or, I can coordinate with VoltronData or @EpsilonPrime on some sort of shared maintainership.

**Comment by drin** [comment]
with a very brief look at the code, it seems like the code doesn't need to be compiled in-tree, but let me know if that's incorrect.

**Comment by FelixYBW** [comment]
> Is velox open to someone else maintaining the code in a separate repository as @EpsilonPrime mentioned?
> 

Definitely. You may contribute if you need. The code in velox is uncompleted yet. You may refer to the one Gluten currently is using:
https://github.com/oap-project/gluten/tree/main/cpp/velox/substrait

The substrait we used is a pretty old version, you may upgrade to latest version. Also substrait itself is not completed yet. There should be some PR to add functions.

**Comment by mbasmanova** [comment]
@drin 

>  Is velox open to someone else maintaining the code in a separate repository as @EpsilonPrime mentioned?

Sure. That works. Thanks.

**Comment by drin** [comment]
> Your contribution should be to Velox directly.

> If the conversion in Velox is mature enough

To clarify: it is preferred that I just help maintain the substrait conversion in the velox repo itself?

I was considering creating a velox-substrait repo and porting most of the [CMakeLists.txt](https://github.com/oap-project/gluten/blob/main/cpp/velox/CMakeLists.txt) from gluten to place in the new repo. Then, the code can be compiled out of the velox source tree and any systems that use velox can leverage that as a submodule or something similar. However, it sounds like this is not the preferred approach?

Would a preferred approach be to somehow facilitate changes upstreamed from gluten into velox?

The reason I ask is that it seems that if gluten is willing to use substrait conversion in the velox repo, but they're maintaining their own version already, what is the barrier to them just migrating the code themselves and working in the velox repo?

I also want to clarify that I am currently a grad student that is not funded by anything gluten or velox related, so if some of the need is for someone to review changes to upstream into velox then I probably would not be able to keep up with that.

**Comment by mbasmanova** [comment]
@drin 

> I was considering creating a velox-substrait repo 

A separate repo would be preferred. Thanks.

**Comment by drin** [comment]
thanks for the fast response!

**Comment by drin** [comment]
I'll try to have https://github.com/drin/velox-substrait/issues/1 finished soon so that you can resolve this issue.

**Comment by mbasmanova** [comment]
@drin Thanks.

**Comment by FelixYBW** [comment]
> The reason I ask is that it seems that if gluten is willing to use substrait conversion in the velox repo, but they're maintaining their own version already, what is the barrier to them just migrating the code themselves and working in the velox repo?

Because upstreaming to velox still needs much clean up work and take much time. It's not Gluten's high priority work.

If you are going to setup a new repo, let's remove the code in velox now.

**Comment by FelixYBW** [comment]
@mbasmanova Any possible to put the substrait-Velox repo as a side project of Velox?

**Comment by mbasmanova** [comment]
@FelixYBW Binwei, what do you mean by "side project"?

**Comment by FelixYBW** [comment]
I mean put the repo like https://github.com/facebookincubator/substrait-velox

**Comment by mbasmanova** [comment]
@FelixYBW I don't think we can create a repo like that without maintaining it.

**Comment by drin** [comment]
@FelixYBW do you want to maintain the repo in a separate org? maybe we can ask to keep it in https://github.com/substrait-io/?

**Comment by drin** [comment]
I'm asking in the substrait slack as well (substrait.slack.com)

**Comment by drin** [comment]
The initial responses seem to be the following:

> I'd like to see at least one committer and one non-committer that are invested in maintaining such a project

and

> Presumably a new repo would require an SMC vote

So, I think for a short-term solution either me owning it or making an "incubation" organization is the most timely solution
